From ba60a1acdc6404351eb12757f2b9f1253df20869 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm.code@gmail.com>
Date: Tue, 15 Jan 2019 14:10:28 -0800
Subject: [PATCH] fsck_msdos: Build static lib for recovery

 * Add libfsck_msdos for recovery multi-call driver.

 * Rename ask to msdos_ask to avoid conflict with libe2fsk.

Change-Id: I3deca5739f99dce2f0f2e861e46f06d6d7f5b582
---
 Android.mk | 23 +++++++++++++++--------
 ext.h      |  2 ++
 2 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/Android.mk b/Android.mk
index 8dec83a..9c48cb1 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,12 +1,8 @@
 LOCAL_PATH := $(call my-dir)
 
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := boot.c check.c dir.c fat.c main.c
-
-LOCAL_C_INCLUDES := external/fsck_msdos/
-
-LOCAL_CFLAGS := -O2 -g \
+common_src_files := boot.c check.c dir.c fat.c main.c
+common_includes := external/fsck_msdos
+common_cflags := -O2 -g \
     -Wall -Werror \
     -D_BSD_SOURCE \
     -D_LARGEFILE_SOURCE \
@@ -16,8 +12,19 @@ LOCAL_CFLAGS := -O2 -g \
     -Wno-format \
     -Wno-sign-compare
 
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := $(common_src_files)
+LOCAL_C_INCLUDES := $(common_includes)
+LOCAL_CFLAGS := $(common_cflags)
 LOCAL_MODULE := fsck_msdos
 LOCAL_MODULE_TAGS :=
 LOCAL_SYSTEM_SHARED_LIBRARIES := libc
-
 include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := $(common_src_files)
+LOCAL_C_INCLUDES := $(common_includes)
+LOCAL_CFLAGS := $(common_cflags) -Dmain=fsck_msdos_main
+LOCAL_MODULE := libfsck_msdos
+LOCAL_MODULE_TAGS :=
+include $(BUILD_STATIC_LIBRARY)
diff --git a/ext.h b/ext.h
index 6d183e9..052db22 100644
--- a/ext.h
+++ b/ext.h
@@ -52,6 +52,8 @@ extern int skipclean;	/* skip clean file systems if preening */
 
 extern struct dosDirEntry *rootDir;
 
+#define ask msdos_ask
+
 /*
  * function declarations
  */
-- 
2.17.1

